bamboolean/lexer.py

```diff
@@ -32,38 +36,37 @@ def tree_repr(self):
 
 
 class Lexer:
-    def __init__(self, text):
-        assert isinstance(text, str)
+    def __init__(self, text: str) -> None:
         self.text = text
         self.position = 0
         self.current_char = self.text[self.position] if self.text else None
 
-    def error(self):
+    def error(self) -> NoReturn:
         raise BambooleanLexerError(
             ("Error tokenizing input on character: "
              "{} and position: {}.\nExpr: {}".format(
                  self.current_char, self.position, self.text))
         )
 
-    def _is_eof(self, pos):
+    def _is_eof(self, pos: int) -> bool:
         return pos > len(self.text) - 1
 
-    def next(self):
+    def next(self) -> None:
```

Comment:
Name `next` is misleading since there is a `next` function for iterators, that does return value